src/Apps/Juno/JsonTypes.hs

```diff
@@ -242,5 +244,54 @@ instance FromJSON TransactRequest where
                                            <*> v .: "digest"
     parseJSON _ = mzero
 
--- test
---txBS = "{\"payload\":{\"data\":\"mybody\",\"code\":\"mycode\"},\"digest\":{\"hash\":\"hashy\",\"key\":\"key\"}}"
+-- {
+-- "payload": {
+--   "cmds": ["string"]
+-- },
+-- "digest": {
+--   "hash": "string",
+--   "key": "string"
+-- }
+data CommandBatch = CommandBatch {
+      _batchCmds :: [Text]
+    } deriving (Eq,Show,Ord)
+
+instance ToJSON CommandBatch where
+    toJSON (CommandBatch cmds) = object ["cmds" .= cmds]
+instance FromJSON CommandBatch where
+     parseJSON (Object v) = CommandBatch <$> v .: "cmds"
+     parseJSON _ = mzero
+
+data CommandBatchRequest = CommandBatchRequest {
+      cmdBatchpayload :: CommandBatch,
+      cmdBatchdigest :: Digest
+    } deriving (Show, Generic, Eq)
+
+instance ToJSON CommandBatchRequest where
+  toJSON (CommandBatchRequest payload' digest') = object ["payload" .= payload', "digest" .= digest']
+instance FromJSON CommandBatchRequest where
+  parseJSON (Object v) = CommandBatchRequest <$> v .: "payload"
+                                             <*> v .: "digest"
+  parseJSON _ = mzero
+
+mJsonBsToCommand :: BLC.ByteString -> Maybe BSC.ByteString
```

Comment:
Spent some time on this as I don't like repeatedly doing the BS -> Value conversion, but doing it once means breaking out the `parseJSON` calls. I figured it out though:
```
mJsonBsToCommand :: BLC.ByteString -> Maybe BSC.ByteString
mJsonBsToCommand bs = JSON.decode bs >>= \v ->
                      (mAdjustAccount <$> parseMaybe parseJSON v) <|>
                      (mAdjustCreateAcct <$> parseMaybe parseJSON v) <|>
                      (mtransact <$> parseMaybe parseJSON v)
  where
    mAdjustAccount (AccountAdjustPayload acct amt) =
        BSC.pack $ "AdjustAccount " ++  T.unpack acct ++ " " ++ show (toRational amt)

    mAdjustCreateAcct (AccountPayload acct) =
        BSC.pack $ "CreateAccount " ++  T.unpack acct

    mtransact (TransactBody code _) =
        BSC.pack $ T.unpack code
```

I assume this is more-or-less throwaway code [we really need to directly submit these instead of converting them back into a BS] but might as well avoid super-smelly stuff.